examples/adjoints/adjoints_ex6/adjoints_ex6.C

```diff
@@ -0,0 +1,526 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2015 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+// <h1>Adjoints Example 6 - Poisson Equation in square domain with AdjointRefinementErrorEstimator</h1>
+// \author Vikram Garg
+// \date 2017
+//
+// This example solves the Poisson equation, whose solution displays a sharp layer,
+// with QoI based adjoint error estimation and adaptive mesh refinement. The exact
+// solution is . This example also illustrates the use of the adjoint Dirichlet boundary
+// condition capability, necesary for handling flux QoIs. Since we use the adjoint capabilities
+// of libMesh in this example, we use the DiffSystem framework. This file (adjoints_ex6.C)
+// contains the declaration of mesh and equation system objects, poissonsystem.C
+// contains the assembly of the system. Postprocessing to compute the QoI is done in element_postprocess.C.
+// There is no need for element and side qoi derivative functions, since the adjoint RHS is supplied
+// by the adjoint dirichlet boundary condition.
+
+// WARNING: To maintain consistency, the flux must be computed internally in the library as R^h(u^h, L^h)
+// where R^h is the residual vector, u^h is the solution, and L^h the lift function at the current mesh.
+// It is w.r.t to this QoI evaluation that the adjoint error estimate R^h+([u^h]+, z^h+ - [L^h]+) is a
+// valid estimate. Automatic flux QoI evaluation via Adjoint Dirichlet boundary conditions is forthcoming
+// in libMesh, after which flux QoIs will be computed internally and there will be no need for
+// element_postprocess.C
+
```

Comment:
Ah, wait, here it is.  FEMSystem is the only code that *automatically* handles adjoint DBC based flux QoI evaluation in assemble_qoi(), right?  Any *other* system types should need to handle things consistently, but that should be a completed feature in fem_system.C